src/index.ts

```diff
@@ -1,16 +1,50 @@
+import { Fraction, transitiveOrderbook } from '@gnosis.pm/dex-contracts'
 import express from 'express'
+import Web3 from 'web3'
+import { CategoryServiceFactory, CategoryConfiguration, Category, LogLevel } from 'typescript-logging'
+import { OrderbookFetcher } from './orderbook_fetcher'
+import * as yargs from 'yargs'
+
+const argv = yargs
+  .option('node', {
```

Comment:
That's a cool resource. I read the part about configs and see they suggest to use env vars. They don't however argue why env vars are better than command line arguments.

yargs allows using env vars as overrides for command line arguments (https://github.com/yargs/yargs/blob/master/docs/api.md#envprefix). 

This seems to be the most flexible way. It still allows to easily have all configuration checkable in one place and auto generate a help from the description.

Reply:
If it allows env var, kind of OK, although normally imo, u should go for env vars and use `dotenv`.

You can have the envs in a `.env` file locally for developing, have an `.env.rinkeby` for that environment etc.

Reply:
Can you elaborate why env vars are superior over command line arguments? 
Particularly with `yargs` I find it nice to have input validation, a help message and all configuration in one place rather than relying on global state.

I also found using checked in `.env` file for deployments (such as `.env.rinkeby`) less practical than specifying the vars directly in the kubernetes config as any adjustment would have to go through an entire PR/merge/build/push cycle rather than just a `kubectl apply`. 

It might make sense in setups where you keep adding config variables that are different in rinkeby and mainnet and you want to avoid updating your kubernetes on every PR that adds or changes ENV vars. 

In our case, I think the only variable thing is the RPC endpoint, which we probably don't want to expose in a public env file in the first place.

Reply:
It's fine Felix :) 

Normally `env` are using to try to make the app to run as close as possible in all the environments. The less you have config a deployment, the better. This is why modeling the differences as something they get from the environment is a good practice.
I just thought, this app just need to run, and get some minimal env vars that change between environments, this is why I thought it was suitable.

It's ok, I understand your approach can also ensure there are only minimal differences between environments.

the `.env` files, I meant them for development time. I find them very useful, but if you are comfortable with the CLI approach, 100%, go for it